scripts/tuple_tree_generator/tuple_tree_generator/generators/python.py

```diff
@@ -152,6 +170,55 @@ def type_hint(self, field: StructField) -> str:
     def to_bool(_input: bool) -> str:
         return "True" if _input else "False"
 
+    def _parse_mixins(self) -> dict[str, List[str]]:
+        mixins: dict[str, List[str]] = {}
+        for mixins_path in self.mixins_paths:
+            with open(Path(mixins_path), "r") as mixin_file:
+                mixins_parsed = ast.parse(mixin_file.read())
+                mixin_classes = []
+                for node in mixins_parsed.body:
+                    if isinstance(node, ast.ClassDef):
+                        mixin_classes.append(node.name)
+                    elif isinstance(node, ast.Assign):
+                        for target in node.targets:
+                            mixin_classes.append(cast(ast.Name, target).id)
+                mixins[mixins_path] = mixin_classes
+        return mixins
```

Comment:
Document how a mixin needs to be structured. Here we're making some assumptions that are rather non-obvious to infer from the code.

Reply:
Done